src/client/particles.cpp

```diff
@@ -35,6 +35,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "client.h"
 #include "settings.h"
 #include "profiler.h"
+#include "client/mapblock_mesh.h"
```

Comment:
I'm not very happy with depending on `mapblock_mesh.h` just for this one common function (`final_color_blend`). Consider moving it to a distinct header to be included by both `mapblock_mesh.h` and `particles.h` (perhaps call it something like `client/lighting.h`?)

Reply:
Move it to `client/lighting.h` or where? What is bad in the header being included here?

Reply:
> What is bad in the header being included here?

There is no real logical dependency of particles on mapblock meshes. It makes no sense (and has tangible downsides such as - if done at scale - increasing compile times and polluting the scope) to include various mapblock mesh classes, functions etc. when all you want is this single common function.

Conceptually, both `client/mapblock_mesh.cpp` and `src/client/particles.cpp` should include a common header `client/lighting.h` (feel free to choose a better name) implemented by `client/lighting.cpp` which provides this light calculation function.

That said, I understand that you probably don't want to introduce a new file for a single function (though over time hopefully more common lighting functions would go there), so I could live with leaving this "dirty".

Another thing I considered would be moving this to `src/lighting.(h|cpp)` which seems like it might be more appropriate, though `client/lighting.(h|cpp)` would IMO make slightly more sense since this function is exclusive to the client - the server doesn't need to know about it.

Reply:
It's worth to note `final_color_blend` is used not only by `client/mapblock_mesh.cpp` and `client/particles.cpp`, but also a few other files as `/client/clientenvironment.cpp`, `/client/game/cpp` and `client/content_cao.cpp` and some of them using only that function (or still another light color calculating ones containing in that header). I think it makes sense to move the following 4 lighting functions to a new file: `encode_light`, `get_sunlight_color` and two overloaded `final_color_blend` instead of the only one

Reply:
I've added now a new file `client/light_colors.h` containing the declarations of those four functions which I mentioned in the comment above and `/client/light_colors.cpp` containing their definitions.

Reply:
Makes sense, thanks!